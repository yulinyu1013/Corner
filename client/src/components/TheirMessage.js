import React from 'react';

const displayMedia = (message) => {
  if (
    message.attachments[0]
    && message.attachments[0] !== null
    && message.attachments[0].file !== null
  ) {
    console.log();
    message.attachments[0].file = typeof message.attachments[0].file === 'string' ? message.attachments[0].file : URL.createObjectURL(message.attachments[0].file);
    if (message.attachments[0].file.split('.').pop() === 'jpg' || message.attachments[0].file.split('.').pop() === 'jpeg' || message.attachments[0].file.split('.').pop() === 'png') {
      return (
        <div
          className="message-img"
          style={{
            float: 'left', marginRight: '18px', color: 'white', backgroundColor: '#9D9B9B',
          }}
        >
          <img
            src={message.attachments[0].file}
            alt="message-attachment"
            className="message-image"
            style={{ float: 'left' }}
          />
        </div>
      );
    }
    if (message.attachments[0].file.split('.').pop() === 'mp4' || message.attachments[0].file.split('.').pop() === 'm4v' || message.attachments[0].file.split('.').pop() === 'avi' || message.attachments[0].file.split('.').pop() === 'mov') {
      return (
        <div
          className="message-vid"
          style={{
            float: 'left', marginRight: '18px', color: 'white',
          }}
        >
          <video
            src={message.attachments[0].file}
            width="100%"
            height="auto"
            controls
          >
            <track kind="captions" />
          </video>
        </div>
      );
    }
    if (message.attachments[0].file.split('.').pop() === 'mp3') {
      return (
        <div
          className="message-aud"
          style={{
            float: 'left', marginRight: '18px', color: 'white',
          }}
        >
          <audio
            src={message.attachments[0].file}
            controls
          >
            <track kind="captions" />
          </audio>
        </div>
      );
    }
  }
  return (
    <div
      className="message"
      style={{
        float: 'left', marginRight: '18px', color: 'white', backgroundColor: '#9D9B9B',
      }}
    >
      {message.text}
    </div>
  );
};

const TheirMessage = ({ lastMessage, message }) => {
  let isFirstMessageByUser = true;
  if (!lastMessage) {
    isFirstMessageByUser = true;
  } else {
    isFirstMessageByUser = lastMessage.sender.username !== message.sender.username;
  }
  return (
    <div className="message-row">
      {isFirstMessageByUser && (
        <div className="message-sender-title">
          <div className="message-sender-name">
            <p>{message.sender.username}</p>
          </div>
          <div
            className="message-avatar"
            style={{ backgroundImage: message.sender && `url(${message.sender.avatar})` }}
          />
        </div>
      )}
      <div className="message-content">
        {message.attachments && message.attachments.length > 0 && message.attachments[0] !== null
          ? (
            <div>
              {displayMedia(message)}
            </div>
          )
          : (
            <div className="message" style={{ float: 'left', backgroundColor: 'goldenrod', marginLeft: isFirstMessageByUser ? '4px' : '48px' }}>
              {message.text}
            </div>
          )}
      </div>
    </div>
  );
};

export default TheirMessage;
