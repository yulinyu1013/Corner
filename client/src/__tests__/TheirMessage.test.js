import React from 'react';
import renderer from 'react-test-renderer';
import TheirMessage from '../components/TheirMessage';
import user from '../images/user.png';

const message = {
  sender: {
    username: 'test',
    avatar: {user}
  },
  attachments: [],
  text: 'test',
}

const Theirmessage = {
  sender: {
    username: 'test',
    avatar: {user}
  },
  attachments: [],
  text: 'test',
}

const message2= {
  sender: {
    username: 'test2',
    avatar: {user}
  },
  attachments: [],
  text: '',
}

it('renders TheirMessage correctly when sending only text and same name', () => {
  const tree = renderer.create(<TheirMessage lastMessage={message}  message = {Theirmessage}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders TheirMessage correctly when sending with no text', () => {
  const tree = renderer.create(<TheirMessage lastMessage={message}  message = {message2}/>).toJSON();
  expect(tree).toMatchSnapshot();
});