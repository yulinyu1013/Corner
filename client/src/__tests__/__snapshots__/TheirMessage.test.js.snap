// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TheirMessage correctly when sending only text and same name 1`] = `
<div
  className="message-row"
>
  <div
    className="message-content"
  >
    <div
      className="message"
      style={
        Object {
          "backgroundColor": "goldenrod",
          "float": "left",
          "marginLeft": "48px",
        }
      }
    >
      test
    </div>
  </div>
</div>
`;

exports[`renders TheirMessage correctly when sending with no text 1`] = `
<div
  className="message-row"
>
  <div
    className="message-sender-title"
  >
    <div
      className="message-sender-name"
    >
      <p>
        test2
      </p>
    </div>
    <div
      className="message-avatar"
      style={
        Object {
          "backgroundImage": "url([object Object])",
        }
      }
    />
  </div>
  <div
    className="message-content"
  >
    <div
      className="message"
      style={
        Object {
          "backgroundColor": "goldenrod",
          "float": "left",
          "marginLeft": "4px",
        }
      }
    >
      
    </div>
  </div>
</div>
`;
